inc/Ui/PageConflict.php

```diff
@@ -44,7 +44,7 @@ public function show()
         // create the form
         $form = new Form(['id' => 'dw__editform']);
         $form->addTagOpen('div')->addClass('no');
-        $form->setHiddenField('id', $ID);
+        $form->setHiddenField('id', $INFO['id']);
```

Comment:
I observe that you use `$INFO['id']` consistently. I was not aware that this was needed. So out of interest, could you explain why you prefer to use `$INFO['id']`? I know that during rendering of a piece of wiki text snippet, some renderers change $ID temporary as well. This could influence the diff and revision retrievement at that moment as well. But is that not intended at that moment? Or is that more the difference between the UI part here and the Changelog section.

Reply:
@Klap-in Thanks for the question. I have tried to use `$INFO` instead of `$ID` as far as reasonable.
I think `$INFO['id']` and `$ID` have same value in Ui stage, it is enough to have global access to `$INFO` array in any methods of Ui classes. I guess we will able to obsolete global `$ID` when most functions are encapsulated in the classes, it is not scope of this PR. My basic idea of refactoring is to reduce number of global symbols (variables, functions) as possible.

AFAIK, `$ID` is a copy of `$INFO['id']` not alias, and is allowed to change the value temporary when it is necessary to build valid **instruction** cache of the page during parsing stage. If the metadata or xhtml may depend on the page id, the instruction data should contain the page id. This is likely related with #3267.